File: Patterns/Facade Pattern/Cpp/HomeTheater/PopcornPopper.cpp
#pragma once

#include <iostream>
#include <string>

using namespace std;

class PopcornPopper {
private:
    string description;

public:
    PopcornPopper(string description) : description{description} {}
    
    void on() {
        cout << description << " on\n";
    }

    void off() {
        cout << description << " off\n";
    }

    void pop() {
        cout << description << " popping popcorn\n";
    }

    string toString() {
        return description;
    }
};

File: Patterns/Facade Pattern/CSharp/HomeTheater/PopcornPopper.cs
using System;

namespace HomeTheater
{
    public class PopcornPopper
    {
        private string description;

        public PopcornPopper(string description)
        {
            this.description = description;
        }

        public void On() => Console.WriteLine($"{description} on");

        public void Off() => Console.WriteLine($"{description} off");

        public void Pop() => Console.WriteLine($"{description} popping popcorn");

        public string Description { get => description; private set => description = value; }
    }
}